src/controllers/groupe.controller.ts

```diff
@@ -0,0 +1,83 @@
+import { NextFunction, Request, Response } from 'express';
+import { Group } from '@prisma/client';
+import { CreateGroupDto } from '@dtos/group.dto';
+import groupService from '@services/groups.service';
+
+class GroupController {
+  public groupService = new groupService();
+
+  public getGroups = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const findAllGroupsData: Group[] = await this.groupService.findAllGroup();
+
+      res.status(200).json({ data: findAllGroupsData, message: 'findAll' });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public getGroupById = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
```

Comment:
We can also use a class to catchAll Async Error (Async Handler) and wrap it around each controller function. 

As as example: 

```
import { NextFunction, RequestHandler, Request, Response } from 'express';
import * as core from 'express-serve-static-core';

export const catchAll =
  <T>(fn: RequestHandler<core.ParamsDictionary, any, T, core.Query, Record<string, any>>) =>
  (req: Request<core.ParamsDictionary, any, T, core.Query, Record<string, any>>, res: Response, next: NextFunction) => {
    Promise.resolve(fn(req, res, next)).catch((err) => {
      next(err);
    });
  };
```


Then the code will update to: 
`catchAll(async(req, res, next): Promise<void> => {})
`